cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala

```diff
@@ -441,16 +836,287 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
+          |In the example, `LetAntiSemiApply` will be used to check for the absence of
+          |the `FRIENDS_WITH` relationship from each person.""".stripMargin,
+      queryText =
+        """MATCH (other:Person)
+          |WHERE NOT ((other)-[:FRIENDS_WITH]->(:Person)) OR (other)-[:WORKS_IN]->(:Location)
+          |RETURN other.name""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("LetAntiSemiApply"))
+    )
+  }
+
+  @Test def selectOrSemiApply() {
+    profileQuery(
+      title = "Select Or Semi Apply",
+      text =
+        """The `SelectOrSemiApply` operator tests for the existence of a pattern predicate and evaluates a predicate.
+          |This operator allows for the mixing of normal predicates and pattern predicates
+          |that check for the existence of a pattern.
+          |First, the normal expression predicate is evaluated, and, only if it returns `false`, is the costly pattern predicate evaluated.""".stripMargin,
+      queryText =
+        """MATCH (other:Person)
+          |WHERE other.age > 25 OR (other)-[:FRIENDS_WITH]->(:Person)
+          |RETURN other.name""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("SelectOrSemiApply"))
+    )
+  }
+
+  @Test def selectOrAntiSemiApply() {
+    profileQuery(
+      title = "Select Or Anti Semi Apply",
+      text =
+        """The `SelectOrAntiSemiApply` operator tests for the absence of a pattern predicate and evaluates a predicate.""".stripMargin,
+      queryText =
+        """MATCH (other:Person)
+          |WHERE other.age > 25 OR NOT (other)-[:FRIENDS_WITH]->(:Person)
+          |RETURN other.name""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("SelectOrAntiSemiApply"))
+    )
+  }
+
+  @Test def conditionalApply() {
+    profileQuery(
+      title = "Conditional Apply",
+      text =
+        """The `ConditionalApply` operator checks whether a variable is not `null`, and if so, the right child operator will be executed.""".stripMargin,
+      queryText =
+        """MERGE (p:Person {name: 'Andres'})
+          |ON MATCH SET p.exists = true""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("ConditionalApply"))
+    )
+  }
+
+  @Test def antiConditionalApply() {
+    profileQuery(
+      title = "Anti Conditional Apply",
+      text =
+        """The `AntiConditionalApply` operator checks whether a variable is `null`, and if so, the right child operator will be executed.""".stripMargin,
+      queryText =
+        """MERGE (p:Person {name: 'Andres'})
+          |ON CREATE SET p.exists = true""".stripMargin,
+      assertions = (p) => assertThat(p.executionPlanDescription().toString, containsString("AntiConditionalApply"))
+    )
+  }
+
```

Comment:
`RollUpApply`

The RollUpApply is used to execute expressions that take a pattern and returns a list with content from the matched pattern, e.g. pattern expressions or pattern comprehension.

queryTest = "MATCH (p:Person) RETURN p.name, [ (p)-[:WORKS_IN]->(location) | location.name ] as cities"

Reply:
👍